riposte-core/src/test/java/com/nike/riposte/server/handler/RoutingHandlerTest.java

```diff
@@ -182,4 +196,99 @@ public void findSingleEndpointForExecution_throws_MultipleMatchingEndpointsExcep
         // when
         handlerSpy.findSingleEndpointForExecution(requestInfoMock);
     }
+
+    @Test
+    public void doChannelRead_HttpRequest_throws_exception_when_content_length_header_greater_than_configured_global_request_limit() {
+        // given
+        doReturn(null).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 10;
+        httpHeaders.set(CONTENT_LENGTH, 100);
+        httpHeaders.set(CONTENT_TYPE, APPLICATION_JSON);
+        handlerSpy = spy(new RoutingHandler(endpoints, maxRequestSizeInBytes));
+
+        // when
+        Throwable thrownException = Assertions.catchThrowable(() -> handlerSpy.doChannelRead(ctxMock, msg));
+
+        // then
+        assertThat(thrownException).isExactlyInstanceOf(TooLongFrameException.class);
+        assertThat(thrownException.getMessage()).isEqualTo("Content-Length header value exceeded configured max request size of 10");
+    }
+
+    @Test
+    public void doChannelRead_HttpRequest_endpoint_overridden_max_request_size_throws_exception() {
+        // given
+        doReturn(100).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 99;
```

Comment:
Should probably test when the global is above and below the endpoint override (to prove that the override is used no matter what). I'd recommend using a `@DataProvider` to set `maxRequestSizeInBytes` to 99 and 101 - the rest of the test should still work as-is I think.